socorro/processor/rules/breakpad.py

```diff
@@ -63,188 +60,73 @@ def action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
-
-    def _execute_external_process(self, crash_id, command_line, processor_meta):
-        # Tokenize the command line into args
-        command_line_args = shlex.split(command_line, comments=False, posix=True)
-
-        # Execute the command line sending stderr (debug logging) to devnull and
-        # capturing stdout (JSON blob of output)
-        subprocess_handle = subprocess.Popen(
-            command_line_args,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.DEVNULL,
+            current_mapping = current_mapping[key_fragment]
+        except KeyError:
+            current_mapping[key_fragment] = {}
+            current_mapping = current_mapping[key_fragment]
+    current_mapping[key_parts[-1]] = value
+
+
+def execute_external_process(crash_id, command_pathname, command_line,
+                             processor_meta, interpret_output):
+    # Tokenize the command line into args
+    command_line_args = shlex.split(command_line, comments=False, posix=True)
+
+    # Execute the command line sending stderr (debug logging) to devnull and
+    # capturing stdout (JSON blob of output)
+    subprocess_handle = subprocess.Popen(
+        command_line_args,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.DEVNULL,
+    )
+    with closing(subprocess_handle.stdout):
+        external_command_output = interpret_output(
+            fp=subprocess_handle.stdout,
+            processor_meta=processor_meta,
+            command_pathname=command_pathname
         )
-        with closing(subprocess_handle.stdout):
-            external_command_output = self._interpret_external_command_output(
-                subprocess_handle.stdout,
-                processor_meta
-            )
 
-        return_code = subprocess_handle.wait()
-        return external_command_output, return_code
-
-    @staticmethod
-    def dot_save(a_mapping, key, value):
-        if '.' not in key or isinstance(a_mapping, DotDict):
-            a_mapping[key] = value
-            return
-        current_mapping = a_mapping
-        key_parts = key.split('.')
-        for key_fragment in key_parts[:-1]:
-            try:
-                current_mapping = current_mapping[key_fragment]
-            except KeyError:
-                current_mapping[key_fragment] = {}
-                current_mapping = current_mapping[key_fragment]
-        current_mapping[key_parts[-1]] = value
-
-    def _save_results(
-        self,
-        external_command_output,
-        return_code,
-        raw_crash,
-        processed_crash,
-        processor_meta
-    ):
-        # FIXME(willkg): we could replace this with glom
-        self.dot_save(processed_crash, self.config.result_key, external_command_output)
-        self.dot_save(processed_crash, self.config.return_code_key, return_code)
-
-    def action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
-        crash_id = raw_crash['uuid']
-        command_parameters = dict(self.config)
```

Comment:
This is how `kill_timeout` and `command_pathname` got into the parameters for interpolation.